tests/func/test_repo.py

```diff
@@ -1,71 +1,60 @@
-        )
-        self.dvc.scm.add([".gitignore", "buzz.dvc"])
-        self.dvc.scm.commit("add buzz")
-        self.dvc.scm.checkout("master")
-
-    def _check(self, branch, target, with_deps, expected):
-        if branch:
-            self.dvc.tree = GitTree(self.dvc.scm.repo, branch)
-        else:
-            self.dvc.tree = WorkingTree()
-        result = self.dvc.collect(target + ".dvc", with_deps=with_deps)
-        self.assertEqual([[str(j) for j in i.outs] for i in result], expected)
-        return result
-
-    def test(self):
-        self._check("", self.BAR, True, [[self.FOO], [self.BAR]])
-        self._check("master", self.BAR, True, [[self.FOO], [self.BAR]])
-        self._check(
-            "new_branch", "buzz", True, [[self.FOO], [self.BAR], ["buzz"]]
+from dvc.system import System
+from dvc.utils.compat import fspath
+
+
+def test_destroy(tmp_dir, dvc):
+    dvc.config.set("cache", "type", "symlink")
+
+    tmp_dir.dvc_gen("file", "text")
+    tmp_dir.dvc_gen({"dir": {"file": "lorem", "subdir/file": "ipsum"}})
+
+    dvc.destroy()
+
+    # Remove all the files related to DVC
+    for path in [".dvc", "file.dvc", "dir.dvc"]:
+        assert not (tmp_dir / path).exists()
+
+    # Leave the rest of the files
+    for path in ["file", "dir/file", "dir/subdir/file"]:
+        assert (tmp_dir / path).is_file()
+
+    # Make sure that data was unprotected after `destroy`
+    for path in ["file", "dir", "dir/file", "dir/subdir", "dir/subdir/file"]:
+        assert not System.is_symlink(fspath(tmp_dir / path))
+
+
+def test_collect(tmp_dir, scm, dvc, run_copy):
+    def collect_outs(*args, **kwargs):
+        return set(
+            str(out.path_info)
+            for stage in dvc.collect(*args, **kwargs)
+            for out in stage.outs
         )
-        result = self._check("new_branch", "buzz", False, [["buzz"]])
-        self.assertEqual([str(i) for i in result[0].deps], ["bar"])
 
+    tmp_dir.dvc_gen("foo", "foo")
+    run_copy("foo", "bar")
+    assert collect_outs("bar.dvc", with_deps=True) == {"foo", "bar"}
+
+    run_copy("bar", "buzz")
+    assert collect_outs("buzz.dvc", with_deps=True) == {"foo", "bar", "buzz"}
+    assert collect_outs("buzz.dvc", with_deps=False) == {"buzz"}
 
-class TestIgnore(TestDvcGit):
-    def _stage_name(self, file):
-        return file + Stage.STAGE_FILE_SUFFIX
 
-    def test_should_not_gather_stage_files_from_ignored_dir(self):
-        ret = main(["add", self.FOO, self.BAR, self.DATA, self.DATA_SUB])
-        self.assertEqual(0, ret)
+def test_stages(tmp_dir, dvc):
+    def stages():
+        return set(stage.relpath for stage in Repo(str(tmp_dir)).stages)
 
-        stages = self.dvc.stages
-        self.assertEqual(4, len(stages))
+    tmp_dir.dvc_gen({"file": "a", "dir/file": "b", "dir/subdir/file": "c"})
 
-        self.create(DvcIgnore.DVCIGNORE_FILE, self.DATA_DIR)
+    assert stages() == {
+        os.path.join("file.dvc"),
```

Comment:
is there need for `os.path.join` here?

Reply:
no need, but just to keep it consistent :